### DISCUSSION TITLE
SSIM for non-square images?

### DISCUSSION THREAD

**Discussion by ljeonjko** [francois-rozet/piqa#34] [Q&A] [answered]
Hi!

Is it possible to run SSIM for non-squared images (e.g. 4x10 pixels)? If impossible, what changes to ssim.py should be made?

**Comment by francois-rozet** [comment, marked as answer]
Hello @ljeonjko,

Yes SSIM can be applied to non-square images. No changes to the source code is necessary.

```python
import torch
import piqa

x = torch.rand(1, 3, 42, 69)
y = torch.rand_like(x)

ssim = piqa.SSIM()
ssim(x, y)  # works as expected
```

However, an 11 by 11 convolution kernel is used in SSIM, which constraints images to have more than 11 pixels in all dimensions.

**Reply by ljeonjko** [reply]
Thanks! I supposed reducing the kernel size should not affect the SSIM too much, right?

**Reply by francois-rozet** [reply]
No, it will affect the result significantly. If you have 4 by 10 "images", I don't think using image quality metrics like SSIM still makes sense. They are usually designed to match human quality perceptions, which fail at such low resolution. Instead you can use PSNR or simply the (R)MSE.

**Reply by ljeonjko** [reply]
Huh, thanks for pointing this out. I'll try both and see what works better. Thanks!
